File: 프로그래머스/lv1/12930. 이상한 문자 만들기/이상한 문자 만들기.java
class Solution {
    public String solution(String s) {
        StringBuilder answer = new StringBuilder();
        boolean isUpper = true;
        for (char c : s.toCharArray()) {
            if (c == ' '){
                answer.append(c);
                isUpper = true;
            } else {
                answer.append(isUpper ? Character.toUpperCase(c) : Character.toLowerCase(c));
                isUpper = !isUpper;
            }
        }
        return answer.toString();
    }
}

File: JavaScript/프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.js
function solution(s) {
    let isUpper = true;
    let answer = [];
    
    s.split('').forEach(c => {
        if (c === " ") {
            isUpper = true;
            answer.push(c);
        } else {
            answer.push(isUpper ? c.toUpperCase() : c.toLowerCase());
            isUpper = !isUpper;
        }
    });
    
    return answer.join('');
}